app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt

```diff
@@ -76,6 +171,7 @@ data class ChatUiState(
             user1ReadMostRecentMessage = false,
             user2ReadMostRecentMessage = false,
             messages = mutableListOf()),
+    val opponentId: String? = null, // To be deleted once VM is integrated with UI
```

Comment:
Is there a reason why it needs to be removed?

`opponentId` from the uiState is only used in the UI once in line 152. If it's not stored in the uiState, then the viewmodel needs to be passed in as a parameter so `viewModel.opponentId` instead of `uistate.opponentId`. 

The UiState is already passed in as a parameter so might be easiest to keep it in the UI. what do u think?

Reply:
We should remove opponentId from ChatUiState to avoid redundancy. It's already accessible through viewModel.opponentId, making it unnecessary to track the same value in two places. This approach reduces complexity and potential errors. If adjusting the UI references from uiState.opponentId to viewModel.opponentId requires significant changes, we can evaluate the impact. However, streamlining this should generally be manageable.

Reply:
True, realised that now, thanks :)